### ISSUE TITLE
monero-wallet-rpc does not appear to store the tx private keys by default

### ISSUE THREAD

**Issue by plavirudar** [monero-project/monero#4316]
As a result of the title, it would therefore appear that a user who is using `monero-wallet-rpc` to send transactions and does not manually and explicitly enable the `get_tx_keys` option in the `sweep_all` or `transfer` methods would lose all of their tx private keys and therefore will not be able to trustlessly prove that they have sent a payment using a block explorer or wallet tool. 

When I look at the RPC wallet itself, the data does appear to still be stored within the wallet somewhere, since the RPC method `get_transfer_by_txid` does in fact decode the destinations, payment_id, amount and fee. I am however also unable to retrieve the tx private key in order to prove payment. 

Is there a way to retrieve the tx private key from a transaction sent by `monero-wallet-rpc` without the option `get_tx_keys` enabled? If not, will it be possible to enable this option by default to prevent users from irretrievably losing the tx private key?

**Comment by moneromooo-monero** [comment]
Works for me. Did you kill monero-rpc-wallet without saving after transfer ?

**Comment by plavirudar** [comment]
@moneromooo-monero Yes, I had shut down `monero-wallet-rpc` after my transactions in an attempt to retrieve the key from `monero-wallet-cli`, and was unable to restore the tx private key afterwards.

**Comment by moneromooo-monero** [comment]
The get_tx_key RPc.

**Comment by jtgrassie** [comment]
@plavirudar `get_tx_key` is an input parameter for the wallet RPC method `transfer`. Part of the response (if get_tx_key is set to true) is the tx_key. If you then wanted to check this tx in the wallet cli, you would use `check_tx_key <txid> <txkey> <address>`.

**Comment by moneromooo-monero** [comment]
I've also tried with monero-wallet-cli after exiting monero-wallet-rpc, and I also see the tx key (with the get_tx_key command).

**Comment by plavirudar** [comment]
@jtgrassie Yes, my point is that the parameter is set to `false` by default, and does not return the key unless it's explicitly set to `true`, as I have now done. 

@moneromooo-monero Thanks for the update, I have tried the RPC call `get_tx_key` and it does work on the RPC wallet to retrieve the previously assumed lost tx private key. I was unable to find it and assumed such a RPC call did not exist as it was not part of the [official documentation](https://getmonero.org/resources/developer-guides/wallet-rpc.html). 

I had to try a couple of guesses to the param names (without which they returned `"TX ID has invalid format"`), but the params appeared to be `{"txid":"your_tx_id"}`
